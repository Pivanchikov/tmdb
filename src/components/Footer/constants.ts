export const linkName = ['about', 'faq']
export const copyright = 'Â© 2021 TMDb, inc.'
