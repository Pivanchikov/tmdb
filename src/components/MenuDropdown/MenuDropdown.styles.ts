export const styles = theme => ({
  root: {
    display: 'flex',
    alignItems: 'center',
  },
  item: {
    color: theme.palette.primary.main,
  },
  button: { color: theme.palette.secondary.main },
})

export const transform = {
  transformOrigin: 'center top',
}
