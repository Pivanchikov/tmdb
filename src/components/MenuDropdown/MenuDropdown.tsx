import { useState, useRef } from 'react'
import {
  Button,
  ClickAwayListener,
  Grow,
  Paper,
  Popper,
  MenuItem,
  MenuList,
  makeStyles,
} from '@material-ui/core'
import { useTranslation } from 'react-i18next'

import { styles, transform } from './MenuDropdown.styles'

interface Props {
  name: string
  items: Array<string>
}

const useStyles = makeStyles(styles)

export const MenuDropdown = ({ name, items }: Props) => {
  const classes = useStyles()
  const [opened, setOpened] = useState(false)
  const anchorRef = useRef<HTMLButtonElement>(null)
  const { t } = useTranslation()

  const handleToggle = () => setOpened(state => !state)

  return (
    <div className={classes.root}>
      <>
        <Button ref={anchorRef} onClick={handleToggle} className={classes.button}>
          {t(name)}
        </Button>
        <Popper open={opened} anchorEl={anchorRef.current} transition onMouseLeave={handleToggle}>
          {({ TransitionProps }) => (
            <Grow {...TransitionProps} style={transform}>
              <Paper>
                <ClickAwayListener onClickAway={handleToggle}>
                  <MenuList>
                    {items.map((item, index) => (
                      <MenuItem onClick={handleToggle} className={classes.item} key={index}>
                        {t(item)}
                      </MenuItem>
                    ))}
                  </MenuList>
                </ClickAwayListener>
              </Paper>
            </Grow>
          )}
        </Popper>
      </>
    </div>
  )
}
