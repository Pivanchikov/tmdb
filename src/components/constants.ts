export const films = ['popular', 'nowPlaying', 'upcoming', 'topRated']
export const serials = ['popular', 'airingToday', 'onTv', 'topRated']
export const people = ['popularPeople']
export const another = ['support', 'API']
export const burgerItems = {
  top: ['movies', 'tvShows', 'people'],
  middle: ['API', 'support', 'about'],
  bottom: ['signIn', 'signUp'],
}
export const languages = [
  ['Ru', 'ru'],
  ['Gb', 'en'],
]
export const desktopWidth = 1280
export const TIMEOUT_MS = 300
export const QUERY_LENGTH_THRESHOLD = 3
