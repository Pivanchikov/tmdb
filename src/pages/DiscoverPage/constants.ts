export const STEP = 1
export const MIN_YEAR = 1920
export const MAX_YEAR = 2021
