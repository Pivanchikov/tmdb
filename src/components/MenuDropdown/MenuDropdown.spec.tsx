import { render as testingLibraryRender, fireEvent } from '@testing-library/react'

import { MenuDropdown } from './MenuDropdown'
import { films } from '../constants'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(value => value),
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    },
  }),
}))

describe('MenuDropdown', () => {
  const render = () => testingLibraryRender(<MenuDropdown name="MOVIES" items={films} />)

  it('renders correctly initial document', () => {
    const { getByRole, queryByText } = render()

    expect(getByRole('button')).toBeInTheDocument
    expect(queryByText(/Movies/i)).toBeInTheDocument
  })

  it('click on menu button', () => {
    const expectedLength = 4

    const { getByRole, getByText, queryAllByRole } = render()

    fireEvent.click(getByRole('button'))

    expect(queryAllByRole('menuitem')).toHaveLength(expectedLength)
    films.forEach(item => expect(getByText(item)).toBeInTheDocument)

    fireEvent.click(getByRole('button'))
    films.forEach(item => expect(getByText(item)).not.toBeInTheDocument)
  })
  it('matches snapshot', () => {
    const { asFragment } = render()

    expect(asFragment()).toMatchSnapshot()
  })
})
