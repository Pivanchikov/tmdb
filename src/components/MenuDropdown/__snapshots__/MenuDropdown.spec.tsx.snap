// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDropdown matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-7"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-9"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        MOVIES
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
